engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentStatusMapper.scala

```diff
@@ -0,0 +1,55 @@
+package pl.touk.nussknacker.k8s.manager
+
+import io.circe.Json
+import pl.touk.nussknacker.engine.api.{CirceUtil, ProcessVersion}
+import pl.touk.nussknacker.engine.api.deployment.{ProcessState, ProcessStateDefinitionManager, StateStatus}
+import pl.touk.nussknacker.engine.api.deployment.simple.SimpleStateStatus
+import pl.touk.nussknacker.k8s.manager.K8sDeploymentManager.scenarioVersionAnnotation
+import skuber.apps.v1.Deployment
+
+//Based on https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#deployment-status
+class K8sDeploymentStatusMapper(definitionManager: ProcessStateDefinitionManager) {
+
+  private val availableCondition = "Available"
+
+  private val progressingCondition = "Progressing"
+
+  //https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
+  //"For some conditions, True represents normal operation, and for some conditions, False represents normal operation."...
+  //in our case Availability and Progressing have "positive polarity" as described in link above...
+  private val conditionStatusToCheck = "True"
+
+  private[manager] def findStatusForDeployments(deployments: List[Deployment]): Option[ProcessState] = {
+    def status(deployment: Deployment): ProcessState = {
+      val version = deployment.metadata.annotations.get(scenarioVersionAnnotation)
```

Comment:
Maybe we can have both creation of this annotation and extraction in `K8sDeploymentManager` companion object to keep them synchronized?

Reply:
done